compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -326,6 +324,23 @@ BLangStatementExpression desugar(BLangQueryExpr queryExpr, SymbolEnv env,
         return streamStmtExpr;
     }
 
+    private BType getMapType(BType type) {
+        BType refType = Types.getReferredType(type);
+        if (refType.tag == TypeTags.UNION) {
+            for (BType memberType : ((BUnionType) type).getMemberTypes()) {
+                BType tempType = getMapType(memberType);
+                if (tempType != symTable.noType) {
+                    return tempType;
+                }
+            }
+        } else if (refType.tag == TypeTags.INTERSECTION) {
+            return getMapType(((BIntersectionType) refType).effectiveType);
+        } else if (refType.tag == TypeTags.MAP) {
+            return refType;
+        }
+        return symTable.noType;
```

Comment:
When can noType be returned?

Reply:
I guess, it cannot be happened in here. I will change the logic to return `refType` in `else` case.

Reply:
See https://github.com/ballerina-platform/ballerina-lang/pull/36743#discussion_r938450434